src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2390,6 +2403,444 @@ def remove_triggered_webjob(cmd, resource_group_name, name, webjob_name, slot=No
+    return logger.warning("not a webapp")
+
+
+def functionapp_list_vnet_int(cmd, name, resource_group, slot=None):
+    webapp = _generic_site_operation(cmd.cli_ctx, resource_group, name, 'get', slot)
+    if webapp.kind == "functionapp":
+        return list_vnet_int(cmd, name, resource_group, slot)
+    return logger.warning("not a functionapp")
+
+
+def list_vnet_int(cmd, name, resource_group, slot):
+    client = web_client_factory(cmd.cli_ctx)
+    if slot is None:
+        result = list(client.web_apps.list_vnet_connections(resource_group, name))
+    else:
+        result = list(client.web_apps.list_vnet_connections_slot(resource_group, name, slot))
+    mod_list = []
+
+    # reformats the vnet entry, removing unecessary information
+    for x in result:
+        # removes GUIDs from name and id
+        longName = x.name
+        if '_' in longName:
+            usIndex = longName.index('_')
+            shortName = longName[usIndex + 1:]
+        else:
+            shortName = longName
+        v_id = x.id
+        lastSlash = v_id.rindex('/')
+        shortId = v_id[:lastSlash] + '/' + shortName
+        # extracts desired fields
+        certThumbprint = x.cert_thumbprint
+        location = x.additional_properties["location"]
+        v_type = x.type
+        vnet_resource_id = x.vnet_resource_id
+        id_strings = v_id.split('/')
+        resourceGroup = id_strings[4]
+        routes = x.routes
+
+        vnet_mod = {"certThumbprint": certThumbprint,
+                    "id": shortId,
+                    "location": location,
+                    "name": shortName,
+                    "resourceGroup": resourceGroup,
+                    "routes": routes,
+                    "type": v_type,
+                    "vnetResourceId": vnet_resource_id}
+        mod_list.append(vnet_mod)
+
+    return mod_list
+
+
+def webapp_add_vnet_int(cmd, name, resource_group, vnet, subnet, slot=None):
+    webapp = _generic_site_operation(cmd.cli_ctx, resource_group, name, 'get', slot)
+    if webapp.kind == "app":
+        return add_vnet_int(cmd, name, resource_group, vnet, subnet, slot)
+    return logger.warning("not a webapp")
+
+
+def functionapp_add_vnet_int(cmd, name, resource_group, vnet, subnet, slot=None):
+    webapp = _generic_site_operation(cmd.cli_ctx, resource_group, name, 'get', slot)
+    if webapp.kind == "functionapp":
+        return add_vnet_int(cmd, name, resource_group, vnet, subnet, slot)
+    return logger.warning("not a functionapp")
+
+
+def add_vnet_int(cmd, name, resource_group, vnet, subnet, slot):
+    client = web_client_factory(cmd.cli_ctx)
+    vnet_client = network_client_factory(cmd.cli_ctx)
+
+    list_all_vnets = vnet_client.virtual_networks.list_all()
+
+    vnet_id = ''
+    for v in list_all_vnets:
+        if v.name == vnet:
+            vnet_id = v.id
+
+    # parsing the arm uri in order to extract vnet_name and vnet_resource_group
+    vnet_id_strings = vnet_id.split('/')
```

Comment:
this can be error prone - is there no API call we can make to get the Vnet Name & Vnet_RG? Some Relay call may be?